lib/pacemaker/pcmk_sched_instances.c

```diff
@@ -725,31 +725,32 @@ multi_update_interleave_actions(pe_action_t *first, pe_action_t *then,
             const pe_resource_t *then_rsc = NULL;
 
             first_rsc = pcmk__get_rsc_in_container(first_child, node);
-            if ((first_rsc != NULL) && strstr(first->task, "stop")) {
-                /* Except for 'stopped' we should be looking at the
-                 * in-container resource, actions for the child will
-                 * happen later and are therefor more likely to align
-                 * with the user's intent.
+            if ((first_rsc != NULL)
+                && pcmk__str_any_of(first->task, CRMD_ACTION_STOP,
+                                    CRMD_ACTION_STOPPED, NULL)) {
+                /* Use the containerized resource since its actions will happen
+                 * later and are more likely to align with the user's intent.
                  */
                 first_action = find_first_action(first_rsc->actions, NULL,
-                                                 task2text(task), node);
+                                                 first_task, node);
```

Comment:
Interesting that we have the `task2text(clone_child_action(action))` pattern for first and not for then. I don't know if it makes any difference whether we grab `action->task` directly (as in `then`) vs. the roundabout way of getting the enum from the clone child and converting it back to text (as in `first`).

Reply:
Yeah this part is really confusing, some later work helps